src/provider/provider_file_memory.c

```diff
@@ -131,7 +134,11 @@ static umf_result_t file_initialize(void *params, void **provider) {
     umf_file_memory_provider_params_t *in_params =
         (umf_file_memory_provider_params_t *)params;
 
-    size_t page_size = utils_get_page_size();
+    if (in_params->is_fsdax && in_params->visibility != UMF_MEM_MAP_SYNC) {
```

Comment:
nit: Since `UMF_MEM_MAP_SYNC` is always required for FSDAX case, could we implicitly add this visibility flag when `is_fsdax == true`?

Reply:
No, it should be verified if mapping with MAP_SYNC succeeded - I have just implemented that.
Done.